src/lib.rs

```diff
@@ -1329,6 +1367,31 @@ impl<'ctx> Bindings<'ctx> {
         try!(eof(&mut ps.s));
         ps.s.out.flush()
     }
+
+    /// Checks if format_bindings is set and runs rustfmt on the file
+    fn format_generated_file(&self, file: &Path) {
+        if !self.context.options().format_bindings {
+            return;
+        }
+
+        let rustfmt = if let Ok(rustfmt) = which::which("rustfmt") {
+            rustfmt
+        } else {
+            error!("Could not find rustfmt in the global path.");
+            return;
+        };
+
+        let mut cmd = Command::new(rustfmt);
+
+        if let Some(path) = self.context.options().format_configuration_file.as_ref().and_then(
+            |f| f.to_str()) {
+            cmd.args(&["--config-path", path]);
+        }
+
+        if let Err(e) = cmd.arg(file).status() {
```

Comment:
> he problem with the status code is that if rustfmt can not format (because line too long or whatever), it ends with status != 0.

Yeah. Ideally we would have some way of distinguishing between "real" failures and "warnings". I guess we should just add a comment to this effect here and move on with our lives.

cc @nrc: you may find this ^ useful for informing rustfmt development maybe

> I also can not return an error here, because write_to_file returns a io::Result.

`io::Error` allows for custom errors (like in the above snippet) and the `std::process` APIs also return `io::Result`s, so I don't understand the problem. I must not be following something. But I guess it doesn't matter since we aren't changing these things.